### ISSUE TITLE
no auth

### ISSUE THREAD

**Issue by solong2017** [gocd/docker-registry-artifact-plugin#761]
Hi, I want use this plugin for my private dockerhub, when I config this plugin in gocd, I cannot save the configuration without username & password because of I haven't config it in my private dockerhub.
So is it the only way that config auth in my dockerhub to use this plugin?

**Comment by chadlwilson** [comment]
If your question is whether you need to configure authentication details to access a private docker hub, then the answer is yes. You would normally generate a personal access token on Docker hub (with only the necessary access) to use as the 'password'.

**Comment by solong2017** [comment]
No, my private docker hub no need to auth when to access, so I can access it without account & password. So I mean that how to use this plugin in this scene, or is there any way to use the plugin without auth? thank you.

**Comment by chadlwilson** [comment]
If you don't need to auth, then it's not `private`. Private implies authentication and authorization is required for access. What am I missing?

Perhaps you did a `docker login` some time earlier in which case docker is caching your credentials somewhere?

**Comment by solong2017** [comment]
oh, I think we have a misunderstand here, I think that is my fault, my private docker hub means I build on my local server, not the remote server, use docker pull & run registry, it is not need to auth by default.

**Comment by chadlwilson** [comment]
Oh okay, since you mentioned Docker *Hub* I thought you meant the cloud based one since that's its name :)

Currently the credentials are mandatory as for most purposes one would need to authenticate.

I'd suggest you add auth to your local test registry in the meantime?

**Comment by solong2017** [comment]
Ok, I thought there are some releases can support my scene, or maybe I haven't found the right way to use it in this scene, anyway I will add follow your suggestion, thank you.
